src/capymoa/anomaly/online_isolation_forest.py

```diff
@@ -0,0 +1,341 @@
+from __future__ import annotations
+from capymoa.base import AnomalyDetector
+from capymoa.instance import Instance, LabelIndex
+from capymoa.stream._stream import Schema
+from capymoa.type_alias import AnomalyScore
+from concurrent.futures import ThreadPoolExecutor
+from itertools import repeat
+from multiprocessing import cpu_count
+from numpy import argsort, asarray, empty, finfo, inf, log, ndarray, sort, split, vstack, zeros
+from numpy.linalg import norm
+from numpy.random import default_rng, Generator
+from typing import Optional
+
+
+class OnlineIsolationForest(AnomalyDetector):
+    """ Online Isolation Forest
+
+    This class implements the Online Isolation Forest (oIFOR) algorithm, which is
+    an ensemble anomaly detector capable of adapting to concept drift.
+
+    Reference:
+    @inproceedings{Leveni2024,
+    title        = {Online Isolation Forest},
+    author       = {Leveni, Filippo and Weigert Cassales, Guilherme and Pfahringer, Bernhard and Bifet, Albert and Boracchi, Giacomo},
+    booktitle    = {International Conference on Machine Learning (ICML)},
+    year         = {2024},
+    organization = {Proceedings of Machine Learning Research (PMLR)}}
+    Example usage:
+    >>> from capymoa.datasets import ElectricityTiny
+    >>> from capymoa.anomaly import OnlineIsolationForest
+    >>> from capymoa.evaluation import AUCEvaluator
+    >>> stream = ElectricityTiny()
+    >>> schema = stream.get_schema()
+    >>> learner = OnlineIsolationForest(schema=schema)
+    >>> evaluator = AUCEvaluator(schema)
+    >>> while stream.has_more_instances():
+    ...     instance = stream.next_instance()
+    ...     proba = learner.score_instance(instance)
+    ...     evaluator.update(instance.y_index, proba)
+    ...     learner.train(instance)
+    >>> auc = evaluator.auc()
+    >>> print(f"AUC: {auc:.2f}")
+    AUC: 0.52
+    """
+    def __init__(self, schema: Schema = None, random_seed: int = 1, num_trees: int = 32, max_leaf_samples: int = 32,
+                 type: str = 'adaptive', subsample: float = 1.0, window_size: int = 2048, branching_factor: int = 2,
```

Comment:
- `type` and `metric` should be a `typing.Literal` e.g `Literal["fixed", "adaptive]` to document the options. 
- The variable name `type` should be avoided because it overrides a built in function. 
- `metric` is not documented in the docstring